_posts/2019-08-26-how-to-use-gruntwork-infrastructure-as-code-library.adoc

```diff
@@ -640,140 +640,124 @@ next two sections of the guide will walk you through how to use each of these.
 
-.infrastructure-modules/networking/vpc-app/main.tf
-[source,hcl]
-----
-terraform {
-  # Partial configuration for the backend: https://www.terraform.io/docs/backends/config.html#partial-configuration
-  backend "s3" {}
-
-  # Only allow this Terraform version. Note that if you upgrade to a newer version, Terraform won't allow you to use an
-  # older version, so when you upgrade, you should upgrade everyone on your team and your CI servers all at once.
-  required_version = "= 0.12.6"
+variable "cidr_block" {
+  description = "The IP address range of the VPC in CIDR notation. A prefix of /18 is recommended. Do not use a prefix higher than /27. Examples include '10.100.0.0/18', '10.200.0.0/18', etc."
+  type        = string
 }
-----
 
-This configures Terraform as follows:
-
-Configure a backend::
-  The code above configures a _backend_, which is a shared location where Terraform state can be stored and accessed by
-  your team. You can use any of the https://www.terraform.io/docs/backends/types/index.html[supported backends] (the
-  example above uses S3, which is a good choice for AWS users). See
-  https://blog.gruntwork.io/how-to-manage-terraform-state-28f5697e68fa[How to manage Terraform state] for more info.
+variable "num_nat_gateways" {
+  description = "The number of NAT Gateways to launch for this VPC. For production VPCs, a NAT Gateway should be placed in each Availability Zone (so likely 3 total), whereas for non-prod VPCs, just one Availability Zone (and hence 1 NAT Gateway) will suffice."
+  type        = number
+}
 
-Partial configuration::
-  The backend uses a _https://www.terraform.io/docs/backends/config.html#partial-configuration[partial configuration]_,
-  which means most of the backend configuration (e.g., which S3 bucket and path to use) will be specified from outside
-  of the code. You'll see an example of this soon.
+# Plus many more optional variables...
 
-Pin the Terraform version::
-  The code above will ONLY allow you to run it with a specific Terraform version. This is a safety measure to ensure
-  you don't accidentally pick up a new version of Terraform until you're ready. This is important because (a) Terraform
-  is a pre 1.0.0 tool, so even patch version number bumps (e.g., `0.12.6` -> `0.12.7`) are sometimes backwards
-  incompatible or buggy and (b) once you've upgraded to a newer version, Terraform will no longer allow you to deploy
-  that code with any older version. For example, if a single person on your team upgrades to `0.12.7` and runs `apply`,
-  then you'll no longer be able to use the state file with `0.12.6`, and you'll be forced to upgrade everyone on your
-  team and all your CI servers to `0.12.7`. It's best to do this explicitly, rather than accidentally, so we recommend
-  pinning Terraform versions.
+----
 
-==== Use the modules from the Gruntwork Infrastructure as Code Library
+==== Gruntwork Infrastructure as Code Library modules are referenced
```

Comment:
The changes to this section is a bit confusing to me. Specifically, the user doesn't (and shouldn't) need to modify the `module` blocks in the service catalog module to modify the vars.

In fact, the references to library modules are useful for informational purposes, but is actually irrelevant when it comes to using the service catalog module as a user in pure terraform, since it's embedded within the service module (and hence the user can't change it without forking).